src/services/pcn-iptables/src/modules/InterfaceLookup.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019 The Polycube Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../Iptables.h"
+#include "datapaths/Iptables_InterfaceLookup_dp.h"
+
+Iptables::InterfaceLookup::InterfaceLookup(const int &index,
+                                     const ChainNameEnum &chain,
+                                     const int &type, Iptables &outer)
+    : Iptables::Program(iptables_code_interfacelookup, index, chain, outer,
+                        (chain == ChainNameEnum::OUTPUT)
+                            ? ProgramType::EGRESS
+                            : ProgramType::INGRESS) {
+  this->type_ = type;
+
+  wildcard_rule_ = false;
+  wildcard_string_ = "";
+
+  load();
+}
+
+Iptables::InterfaceLookup::InterfaceLookup(
+    const int &index, const ChainNameEnum &chain, const int &type,
+    Iptables &outer, const std::map<uint16_t, std::vector<uint64_t>> &ports)
+    : Iptables::Program(iptables_code_interfacelookup, index, chain, outer,
+                        (chain == ChainNameEnum::OUTPUT)
+                            ? ProgramType::EGRESS
+                            : ProgramType::INGRESS) {
+  this->type_ = type;
+
+  auto it = ports.find(0);
+  if (it == ports.end()) {
+    wildcard_rule_ = false;
+    wildcard_string_ = "";
+    // std::cout << "-- wildcard NO --+" << std::endl;
```

Comment:
I would delete this line (and the two debug messages below as well).

Reply:
Why we should delete debug prints?
I think debug messages are always welcome. Btw I don't like `std::cout` approach, using logger is better, but in this method is not easy to get access to logger instance (maybe a point to consider for developer-friendly improvements)

Reply:
Debug print are always welcome, but... debug means for the developer (this should be avoided in the final commit, as we should commit only code that has been properly tested), of for the user who may have an unexpected behavior?
In any case, I would suggest to use the logger.
So, for the time being I would remove this print, and create maybe a issue saying why logger may not be easy to use, so that we can track this issue.